### PR TITLE
Fix SQL compatibility doc

### PR DESCRIPTION
**Pull request by vladimir-golovchenko**
I hereby agree to the terms of the CLA available at: https://yandex.ru/legal/cla/?lang=en

Changelog category (leave one):
- Documentation (changelog entry is not required)

Description:
- Fixed the wrong version of SQL standard
- Extended section E081 according to SQL:2011
- Added missed section S011 according to SQL:2011.

### CODE REVIEW

**Review by alexey-milovidov — APPROVED** [review]
LGTM

File: docs/en/sql-reference/ansi.md [hunk 1]
```diff
@@ -169,6 +179,7 @@ The following table lists cases when query feature works in ClickHouse, but beha
 | **F471**   | **Scalar subquery values**                                                                                               | **Yes**{.text-success}     |                                                                                                                                                                                           |
 | **F481**   | **Expanded NULL predicate**                                                                                              | **Yes**{.text-success}     |                                                                                                                                                                                           |
 | **F812**   | **Basic flagging**                                                                                                       | **No**{.text-danger}       |                                                                                                                                                                                           |
+| **S011**   | **Distinct data types** | | |
```
Comment (alexey-milovidov) [inline docs/en/sql-reference/ansi.md:182]: Distinct **data types** - what's that?

File: docs/en/sql-reference/ansi.md [hunk 2]
```diff
@@ -77,6 +77,16 @@ The following table lists cases when query feature works in ClickHouse, but beha
...
+| E081-01    | SELECT privilege at the table level | | |
+| E081-02    | DELETE privilege | | |
+| E081-03    | INSERT privilege at the table level | | |
+| E081-04    | UPDATE privilege at the table level | | |
+| E081-05    | UPDATE privilege at the column level | | |
+| E081-06    | REFERENCES privilege at the table level | | |
+| E081-07    | REFERENCES privilege at the column level | | |
+| E081-08    | WITH GRANT OPTION | | |
+| E081-09    | USAGE privilege | | |
+| E081-10    | EXECUTE privilege | | |
```
Comment (alexey-milovidov) [inline docs/en/sql-reference/ansi.md:89]: And what remains?

File: docs/en/sql-reference/ansi.md (hunk 1 above)
Comment (vladimir-golovchenko) [reply docs/en/sql-reference/ansi.md:182]: It is *user-defined types* (for example: https://docs.microsoft.com/en-us/openspecs/sql_standards/ms-tsqliso02/7962ac4f-e1c4-41c6-aa7c-6cc40f9fa7ce).

File: docs/en/sql-reference/ansi.md (hunk 2 above)
Comment (vladimir-golovchenko) [reply docs/en/sql-reference/ansi.md:89]: It looks like all features from 'ISO/IEC 9075-2 Foundation (SQL/Foundation) Table 35 — Feature taxonomy and definition for mandatory features' have presented in doc.

Probably make sense to add the list of features from 'Table 39 — Feature taxonomy for optional features' as done some DBMS: [Postgress supported features](https://www.postgresql.org/docs/current/features-sql-standard.html), [Firebird SQL Conformance](https://firebirdsql.org/en/sql-conformance/), etc.

Example of optional features:
```
58 F262 Extended CASE expression
59 F263 Comma-separated predicates in simple CASE expression
60 F271 Compound character literals
61 F281 LIKE enhancements
62 F291 UNIQUE predicate
63 F301 CORRESPONDING in query expressions
64 F302 INTERSECT table operator
65 F302-01 INTERSECT DISTINCT table operator
..
74 F381-01 ALTER TABLE statement: ALTER COLUMN clause
75 F381-02 ALTER TABLE statement: ADD CONSTRAINT clause
76 F381-03 ALTER TABLE statement: DROP CONSTRAINT clause
..
175 S091 Basic array support
176 S091-01 Arrays of built-in data types
177 S091-02 Arrays of distinct types
178 S091-03 Array expressions
179 S092 Arrays of user-defined types
180 S094 Arrays of reference types
181 S095 Array constructors by query
182 S096 Optional array bounds
183 S097 Array element assignment
...
302 T611 Elementary OLAP operations
303 T612 Advanced OLAP operations
304 T613 Sampling
```
-----

It makes sense to double-check my changes based on official ISO documents.
I have access to 'untrusted' SQL:2011 downloaded from the internet (unfortunately, ISO doesn't provide free access to standard and it should be bought).

**Comment by den-crane** [comment]
E121-02 	ORDER BY columns need not be in select list 	No{.text-danger} 	

select 'a' from system.one order by dummy
Yes

----

E121-03 	Value expressions in ORDER BY clause 	No{.text-danger}

select * from numbers(10) order by -number
Yes

**Pull request merged**